akka-http-core/src/main/scala/akka/http/impl/util/package.scala

```diff
@@ -110,22 +106,25 @@ package util {
       setHandler(byteStringIn, new InHandler {
         override def onPush(): Unit = {
           bytes ++= grab(byteStringIn)
-          pull(byteStringIn)
+          if (bytes.length > maxBytes)
+            failStage(new Exception(s"AggregateBytes received more than the configured maximum $maxBytes bytes of data"))
```

Comment:
I'd propose introducing a special exception for this, `AggregateBytesLimitExceededException`?